openapi/openapi/src/test/scala/endpoints/openapi/CoproductEncodingTest.scala

```diff
@@ -0,0 +1,221 @@
+package endpoints.openapi
+
+import endpoints.algebra
+import endpoints.openapi.model._
+import org.scalatest.freespec.AnyFreeSpec
+
+class CoproductEncodingTest extends AnyFreeSpec {
+
+  trait CoproductEncodingAlg
+      extends algebra.Endpoints
+      with algebra.JsonEntitiesFromSchemas
+      with algebra.JsonSchemasFixtures {
+
+    implicit val fooSchema = Foo.schema.named("Foo")
+
+    val foo = endpoint(get(path / "foo"), ok(jsonResponse[Foo]))
+  }
+
+  object OneOfStrategyDocs
+      extends CoproductEncodingAlg
+      with Endpoints
+      with JsonEntitiesFromSchemas {
+
+    override def coproductEncoding = CoproductEncoding.OneOf
+    val api = openApi(Info("OneOf", "1"))(foo)
+  }
+
+  object OneOfWithBaseRefStrategyDocs
+      extends CoproductEncodingAlg
+      with Endpoints
+      with JsonEntitiesFromSchemas {
+
+    override def coproductEncoding = CoproductEncoding.OneOfWithBaseRef
+    val api = openApi(Info("OneOf", "1"))(foo)
+  }
+
+  "OneOf schema encoding" in {
+    val expectedJsonSchema = ujson.Obj(
+      "discriminator" -> ujson.Obj(
+        "propertyName" -> ujson.Str("type"),
+        "mapping" -> ujson.Obj()
```

Comment:
I think it was already the case before your changes, but I believe we should not produce an empty `mapping`. Could we emit it only if it is not empty?